resources/lib/method.py

```diff
@@ -152,6 +152,28 @@ def _get_preselected_items(string):
             return -1
         return items.index(string)
 
+    def _get_preselected_items(string):
+        if not string:
+            return [-1]
+        try:
+            return [int(string)]
+        except TypeError:
+            return [-1]
+        except ValueError:
+            pass
+        items = _split_items(kwargs.get('preselect') or '')
```

Comment:
The `preselect` kwarg is the value passed as `string` so there is no need to get it again. More importantly though, you should be checking `options` kwarg here not the `preselect` as the point of this block of code is to check if the dialog list actually has items before we spend time looking for the preselect index numbers.